lib/ach/batch.rb

```diff
@@ -51,6 +51,10 @@ def to_ach
       @control.company_identification = @header.company_identification
       @control.originating_dfi_identification = @header.originating_dfi_identification
       @control.batch_number = @header.batch_number
+      if @entries[-1].is_a? ACH::BalancingEntryDetail
+        @control.credit_total = @entries[-1].amount
+        @control.debit_total = @entries[-1].amount
+      end
 
       [@header] + @entries + @addendas + [@control]
     end
```

Comment:
Not super convinced of this but may be somewhat clearer rather than calling `@entries[-1]`. 

```suggestion
      if last_entry.is_a? ACH::BalancingEntryDetail
        @control.credit_total = last_entry.amount
        @control.debit_total = last_entry.amount
      end

      [@header] + @entries + @addendas + [@control]
    end
    
    private
    
    def last_entry
      @last_entry ||= @entries.last
    end
```

Could also just change the `@entries[-1]` to `@entries.last`